src/ipc/capnp/common-types.h

```diff
@@ -16,6 +17,14 @@
 
 namespace ipc {
 namespace capnp {
+//! Construct a ParamStream wrapping data stream with serialization parameters
+//! needed to pass transaction objects between bitcoin processes.
+template<typename S>
+auto Wrap(S& s)
+{
+    return ParamsStream{s, TX_WITH_WITNESS};
```

Comment:
8523d19bb3f8c982c6e01a4635c98cd9ac386dea: I wonder if this causes a problem when handling a block template without a witness commitment.

If so, I might just have stratum v2 templates always set a coinbase witness.

Reply:
> [8523d19](https://github.com/bitcoin/bitcoin/commit/8523d19bb3f8c982c6e01a4635c98cd9ac386dea): I wonder if this causes a problem when handling a block template without a witness commitment.
> 
> If so, I might just have stratum v2 templates always set a coinbase witness.

This just sets a serialization flag called `allow_witness` and `fAllowWitness` in [`SerializeTransaction`](https://github.com/bitcoin/bitcoin/blob/1a41e63575986887ae34993b4433ec711ae0ffbc/src/primitives/transaction.h#L256) that includes witness data when it is present instead of omitting it. It ensures that if you pass a CTransaction parameter over IPC, the value of the parameter will be unchanged after it is passed. I should add a comment explaining this better, since more serialization flags are added here later in #10102, and may have to be added in the future. But I don't think there should be a particular concern for the mining interface.